proposals/2020-07-01_aws-kms-keyring-redesign/change.md

```diff
@@ -0,0 +1,340 @@
+
+The [ESDK for JavaScript](https://github.com/aws/aws-encryption-sdk-javascript/blob/75803ed4d3c8b5e86005108c173941c43e81cb56/modules/kms-keyring/src/kms_keyring.ts)
+uses a _KMS keyring_ namespace for the AWS KMS keyrings.
+The proposal can be implemented in a new _AWS KMS keyring_ namespace.
+This now allows the ESDK for JavaScript to use the same AWS KMS naming as other ESDKs.
+The previous _KMS keyring_ namespace can be deprecated.
+
+Usage examples and documentation MUST be updated for the new keying implementations.
+
+## Guide-level Explanation
+
+The existing [AWS KMS Keyring](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md)
+encrypts and decrypts data using AWS KMS [CMKs](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#master_keys).
+It also supports the concept of an _AWS KMS discovery keyring_
+that is enabled through a derived [_Is Discovery_](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md#is-discovery) property.
+
+The _AWS KMS discovery keyring_ changes the underlying behavior of the AWS KMS keyring.
+It returns the unchanged encryption materials on encryption and _attempts_ to decrypt any encrypted data key,
+as long as it can initialize an AWS SDK KMS service client (AWS SDK client).
+
+To communicate with AWS KMS, the ESDK’s AWS KMS keyring needs a way to initialize AWS SDK clients.
+Current implementations use a [client supplier](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md#client-supplier)
+that initializes AWS SDK clients.
+AWS SDK clients are supplied to the AWS KMS keyring at the keyring’s API runtime.
+Client suppliers maintain logic for limiting communication to specific AWS regions and caching AWS SDK clients.
+
+To reduce the scope of the AWS KMS keyring,
+we are moving the keyring’s functionality into multiple smaller-scoped keyrings.
+These smaller-scoped keyrings are configured by _keyring-producing operations_.
+Depending on the implementation, keyring-producing operations can be thought of as factory methods/builders.
+They initialize keyrings.
+
+Keyring-producing operations act as the customer-facing APIs.
+Like client suppliers, they initialize AWS SDK clients.
+However, AWS SDK clients are now initialized before a keyring is configured.
+Keyring-producing operations also handle the load of manually initializing multiple, smaller-scoped keyrings.
+They initialize all keyrings required to meet a specific customer use case.
+We refer to the keyrings that keyring-producing operations produce as _derived keyrings_.
+If a use case involves multiple AWS regions or multiple key names,
+the keyring-producing operation will initialize a [multi-keyring](https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/framework/multi-keyring.md)
```

Comment:
I think it should _always_ be a multi-keyring.
Even if there is only 1 thing involved.
These keyring-producing operations MUST identify the generator,
this makes a single multi-keyring different
then a single AWS KMS keyring.
And can cause surprises.

Reply:
I agree - I'll rephrase to capture this better.